File: 108-convert-sorted-array-to-binary-search-tree/108-convert-sorted-array-to-binary-search-tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func sortedArrayToBST(nums []int) *TreeNode {
    return doWork(nums)
}

func doWork(nums []int) *TreeNode {
    if len(nums) < 1 {
        return nil
    }
    M := int(math.Floor(float64(len(nums)/2)))
    root := TreeNode{nums[M], nil, nil}
    root.Right = doWork(nums[M+1:])
    root.Left = doWork(nums[0:M])
    return &root
}

File: 108-convert-sorted-array-to-binary-search-tree/108-convert-sorted-array-to-binary-search-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {number[]} nums
 * @return {TreeNode}
 */
var sortedArrayToBST = function(nums) {
    return doWork(nums);
};


function doWork(nums){
    if (nums.length < 1){
        return null
    }
    let M = Math.floor(nums.length/2);
    let root = new TreeNode(nums[M]);
    root.left = doWork(nums.slice(0, M));
    root.right = doWork(nums.slice(M+1,nums.length))
    return root;
}